components/brave_wallet/renderer/js_ethereum_provider.cc

```diff
@@ -643,4 +649,128 @@ void JSEthereumProvider::MessageEvent(const std::string& subscription_id,
   FireEvent(ethereum::kMessageEvent, event_args);
 }
 
+void JSEthereumProvider::OnProviderRequested() {
+  AnnounceProvider();
+}
+
+void JSEthereumProvider::BindRequestProviderListener() {
+  v8::Isolate* isolate = blink::MainThreadIsolate();
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context =
+      render_frame()->GetWebFrame()->MainWorldScriptContext();
+
+  auto onProviderRequested = gin::CreateFunctionTemplate(
+      isolate, base::BindRepeating(&JSEthereumProvider::OnProviderRequested,
+                                   weak_ptr_factory_.GetWeakPtr()));
+  auto functionInstance =
+      onProviderRequested->GetFunction(context).ToLocalChecked();
+
+  std::vector<v8::Local<v8::Value>> args;
+  args.push_back(content::V8ValueConverter::Create()->ToV8Value(
+      base::Value("eip6963:requestProvider"), context));
+  args.push_back(std::move(functionInstance));
+
+  CallMethodOfObject(render_frame()->GetWebFrame(), "window",
+                     "addEventListener", std::move(args));
+}
+
+void JSEthereumProvider::AnnounceProvider() {
+  v8::Isolate* isolate = blink::MainThreadIsolate();
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context =
+      render_frame()->GetWebFrame()->MainWorldScriptContext();
+
+  base::Value::Dict provider_info_value;
+  provider_info_value.Set("rDNS", "com.brave.wallet");
+  provider_info_value.Set("uuid", uuid_);
+  provider_info_value.Set("name", "Brave Wallet");
```

Comment:
nit: We may want to support localization of this string since it's user displayed based upon the language within the Browser